### PR TITLE
[EDGEPATRON-131]-Added POST edge api for LOC patron

### PR DESCRIPTION
**Pull request by SinghAdes**

### CODE REVIEW

File: src/main/java/org/folio/edge/patron/PatronHandler.java [hunk 1]
```diff
@@ -28,11 +28,8 @@
 import io.vertx.ext.web.RoutingContext;
 import io.vertx.ext.web.client.HttpResponse;
 import java.text.SimpleDateFormat;
-import java.util.Collections;
-import java.util.Date;
-import java.util.List;
-import java.util.Map;
-import java.util.TimeZone;
+import java.util.*;
```
Comment (gurleenkaurbp) [inline src/main/java/org/folio/edge/patron/PatronHandler.java:31]: Please remove wild card imports

File: src/test/java/org/folio/edge/patron/utils/PatronMockOkapi.java [hunk 2]
```diff
@@ -31,9 +31,7 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.folio.edge.core.utils.test.MockOkapi;
-import org.folio.edge.patron.model.Account;
-import org.folio.edge.patron.model.Charge;
-import org.folio.edge.patron.model.Hold;
+import org.folio.edge.patron.model.*;
```
Comment (gurleenkaurbp) [inline src/test/java/org/folio/edge/patron/utils/PatronMockOkapi.java:34]: Please remove wildcards import.

File: src/main/java/org/folio/edge/patron/PatronHandler.java [hunk 3]
```diff
@@ -151,6 +153,21 @@ public void handlePlaceItemHold(RoutingContext ctx) {
             t -> handleProxyException(ctx, t)));
   }
 
+  public void handlePatronRequest(RoutingContext ctx) {
+    if (ctx.body().asJsonObject() == null) {
+      badRequest(ctx, MSG_HOLD_NOBODY);
```
Comment (Vignesh-kalyanasundaram) [inline src/main/java/org/folio/edge/patron/PatronHandler.java:158]: Please change the error message specific to this api

File: src/main/java/org/folio/edge/patron/PatronHandler.java [hunk 4]
```diff
@@ -259,13 +276,13 @@ protected void handleProxyResponse(RoutingContext ctx, HttpResponse<Buffer> resp
...
     String respBody = resp.bodyAsString();
-    if (logger.isDebugEnabled()) {
+    if (logger.isDebugEnabled() ) {
       logger.debug("response: " + respBody);
     }
 
     String contentType = resp.getHeader(HttpHeaders.CONTENT_TYPE.toString());
 
-    if (resp.statusCode() < 400){
+    if (resp.statusCode() < 400 && Objects.nonNull(respBody)){
```
Comment (Vignesh-kalyanasundaram) [inline src/main/java/org/folio/edge/patron/PatronHandler.java:285]: what is the use of new condition here?

Comment (SinghAdes) [reply src/main/java/org/folio/edge/patron/PatronHandler.java:286]: @Vignesh-kalyanasundaram :- just as precaution , There might be a case when we receive an empty or null response from mod-patron. in this case this code will not fail as NullPointer exception .

File: src/main/java/org/folio/edge/patron/PatronHandler.java (hunk 1 above)
Comment (SinghAdes) [reply src/main/java/org/folio/edge/patron/PatronHandler.java]: fixed

File: src/test/java/org/folio/edge/patron/utils/PatronMockOkapi.java (hunk 2 above)
Comment (SinghAdes) [reply src/test/java/org/folio/edge/patron/utils/PatronMockOkapi.java]: fixed

File: src/main/java/org/folio/edge/patron/PatronHandler.java (hunk 3 above)
Comment (SinghAdes) [reply src/main/java/org/folio/edge/patron/PatronHandler.java]: fixed

**Review by Vignesh-kalyanasundaram — APPROVED** [review]

**Pull request merged**